algorithms/number_theory/eratosthenes_sieve.py

```diff
@@ -0,0 +1,59 @@
+"""Module with the implementation of eratosthenes sieve."""
+
+
+from typing import List
+
+
+class EratosthenesSieve:
+    """Class implementation of the eratosthenes sieve algorithm."""
+
+    def __init__(self, n: int):
+        self._n = n
+        self._no_cousins = [False for _ in range(self._n + 1)]
+
+    def eratosthenes_sieve(self) -> List[bool]:
+        """
+        Eratosthenes sieve algorithm.
+        """
```

Comment:
Can you add a more descriptive docstring here?

At least it should have the specification of the return values.